File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    for i in range(1, total):
        if total%i:
            continue
        j = total//i
        if (i-2)*(j-2) == yellow:
            answer = [j, i]
            break
    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int sum = brown + yellow;
        for(int i = sum - 1; i >= 3; i--) {
            if(sum % i != 0) continue;
            
            int j = sum / i;
            if((i-2)*(j-2) == yellow) {
                answer[0] = i;
                answer[1] = j;
                break;
            }
        }
        
        return answer;
    }
}